btrack/io/hdf.py

```diff
@@ -263,12 +263,25 @@ def filtered_objects(
 
             f_eval = f"x{m['op']}{m['cmp']}"  # e.g. x > 10
 
+            data = None
+
             if m["name"] in properties:
                 data = properties[m["name"]]
-                filtered_idx = [i for i, x in enumerate(data) if eval(f_eval)]
+            elif m["name"] in grp:
+                logger.warning(
+                    f"While trying to filter objects by `{f_expr}` encountered "
+                    "a legacy HDF file."
+                )
+                logger.warning(
+                    "Properties do not persist to objects. Use `hdf.tree()` to "
+                    "inspect the file structure."
+                )
+                data = grp[m["name"]]
             else:
                 raise ValueError(f"Cannot filter objects by {f_expr}")
 
+            filtered_idx = [i for i, x in enumerate(data) if eval(f_eval)]
```

Comment:
May as well remove if not being used?
```suggestion
            filtered_idx = [i for i, _ in enumerate(data) if eval(f_eval)]
```